docs/docs/sample-scripts/spo/create-test-documents-with-openai-api/index.md

```diff
@@ -0,0 +1,114 @@
+        [Parameter(Mandatory = $true, HelpMessage="The OpenAI API key")][string] $OpenAIAPIKey,
+        [Parameter(Mandatory = $false, HelpMessage="Number of test files to generate")][int] $NumberOfDocuments = 5
+    )
+
+    function New-WordFile {
+      param(
+          [string]$FileName,
+          [string]$Content
+      )
+
+      # Load Word application
+      $word = New-Object -ComObject Word.Application
+
+      # Create new document
+      $doc = $word.Documents.Add()
+
+      # Add content to document
+      $selection = $word.Selection
+      $selection.TypeText($Content)
+
+      # Save document
+      $path = Join-Path (Get-Location).Path $FileName
+      $doc.SaveAs("$($path)")
+      $doc.Close()
+
+      # Close Word application
+      $word.Quit()
+
+      # Clean up
+      $null = [System.Runtime.InteropServices.Marshal]::ReleaseComObject([System.__ComObject]$word)
+      [gc]::Collect()
+      [gc]::WaitForPendingFinalizers()
+      Remove-Variable word
+    }
+
+    try {
+      $m365Status = m365 status
+      if ($m365Status -match "Logged Out") {
+        Write-Host "Logging in the User!"
+        m365 login --authType browser
+      }
+
+      $apiEndpoint = "https://api.openai.com/v1/completions"
+
+      $headers = @{
+        "Content-Type" = "application/json"
+        "Authorization" = "Bearer $OpenAIAPIKey"
+      }
+
+      $requestBody = @{
+        prompt = "Generate $NumberOfDocuments comma separated random words"
+        model = "text-davinci-003"        
+        temperature = 0.7
+        max_tokens = 256
+      }
+
+      $response = Invoke-WebRequest -Uri $apiEndpoint -Method POST -Headers $headers -Body (ConvertTo-Json $requestBody)
+      $randomGeneratedWordsContent = $response.Content | ConvertFrom-Json
+      $randomGeneratedWords = $randomGeneratedWordsContent.choices.text -replace "`n","" -replace "`r",""
+      Write-Host "$NumberOfDocuments random generated words are: $randomGeneratedWords"
+      $randomWords = $randomGeneratedWords.Split(',')
+
+      $counter = 1
+      Foreach ($randomWord in $randomWords) {
+        $requestBody = @{
+          prompt = "Write essay on $randomWord"
+          model = "text-davinci-003"        
+          temperature = 1
+          max_tokens = 256
+        }
+
+        $response = Invoke-WebRequest -Uri $apiEndpoint -Method POST -Headers $headers -Body (ConvertTo-Json $requestBody)
+        $essayContent = $response.Content | ConvertFrom-Json
+        $essay = $essayContent.choices.text
+
+        Write-Host "Generating document ($counter / $($randomWords.count)): $randomWord.docx"
+        New-WordFile -FileName "$randomWord.docx" -Content $essay
+
+        Write-Host "Uploading file $randomWord.docx..."
+        $path = Join-Path (Get-Location).Path "$randomWord.docx"
+        m365 spo file add --webUrl $SiteURL --folder $Folder --path $path
+        $counter++
```

Comment:
File upload works great. However, after running the script, my working folder is flooded with randomly generated files, and this was something I didn't really expect/want.
I'm wondering if it would be a good enhancement to permanently remove the generated document from the local system so my working folder stays clean. What do you think?